packages/test/test-end-to-end-tests/src/test/blobs.spec.ts

```diff
@@ -703,28 +688,24 @@ describeCompat("blobs", "NoCompat", (getTestObjectProvider, apis) => {
 	});
 
 	it("reconnection does not block ops when having pending blobs", async () => {
-		const container1 = await provider.makeTestContainer(testContainerConfig);
-		const dataStore1 = (await container1.getEntryPoint()) as ITestDataObject;
-		const runtimeStorage = (container1 as any).runtime.storage;
-
-		let resolveUploadBlob = () => {};
-		const uploadBlobPromise = new Promise<void>((resolve) => {
-			resolveUploadBlob = resolve;
-		});
-
-		const uploadBlobWithDelay = async (target, thisArg, args) => {
-			// Wait for the uploadBlobPromise to be resolved
-			await uploadBlobPromise;
-			const result = Reflect.apply(target, thisArg, args);
-			return result;
-		};
-
-		const delayedUploadBlob = new Proxy(runtimeStorage.createBlob.bind(runtimeStorage), {
-			async apply(target, thisArg, args) {
-				return uploadBlobWithDelay(target, thisArg, args);
+		const uploadBlobPromise = new Deferred<void>();
+		const container1 = await provider.makeTestContainer({
+			...testContainerConfig,
+			loaderProps: {
+				documentServiceFactory: wrapObjectAndOverride(provider.documentServiceFactory, {
+					createDocumentService: {
+						connectToStorage: {
+							createBlob: (dss) => async (blob) => {
```

Comment:
I like how explicit this is about what is being mocked and how

Reply:
@pragya91 if you are still looking to mock storage network calls, you can probably use `wrapObjectAndOverride` that Tony's standardizing on in this PR.